temporalcli/commands_test.go

```diff
@@ -47,13 +50,17 @@ func (h *CommandHarness) Close() {
 	}
 }
 
+// Pieces must appear in order on the line
 func (h *CommandHarness) ContainsOnSameLine(text string, pieces ...string) {
 	// Split into lines, then check each piece is present
 	lines := strings.Split(text, "\n")
 	for _, line := range lines {
 		foundAll := true
+		lastFoundPieceIndex := -1
 		for _, piece := range pieces {
-			if !strings.Contains(line, piece) {
+			if index := strings.Index(line, piece); index > lastFoundPieceIndex {
+				lastFoundPieceIndex = index
+			} else {
```

Comment:
@dandavison - here is update for forcing order

Reply:
Great, thanks for adding that! This looks like it will allow the pieces to match in an overlapping way -- I think we need to set `lastFoundPieceIndex` to the index of the end of `piece`.

Reply:
I think it's ok to be overlapping, it's still in order by start of piece

Reply:
No, it's clear to me that when users make a call like `s.ContainsOnSameLine(out, "Namespace", "default")` they mean

> there exists a line which contains 'Namespace' and 'default', in that order (not overlapping).

I've never seen a test framework that allows them to be overlapping or out of order. I think that your `s.ContainsOnSameLine(out, "Namespace", "default")` call is nice and readable -- nicer to read than a regexp. But how about implementing it as a regular expression match call? That's what I'd do in most languages.

Reply:
Updated